sdk/storage/azure-storage-blob/azure/storage/blob/_shared/authentication.py

```diff
@@ -48,6 +48,33 @@ def _wrap_exception(ex, desired_type):
     # TODO: In the future we will log the trace
     return desired_type('{}: {}'.format(ex.__class__.__name__, msg))
 
+# This method attempts to emulate the sorting done by the service
+@staticmethod
+def _service_sort(input_header):
```

Comment:
Four comments here:
- How about something like `_storage_header_sort`?
- Why is this a `@staticmethod`? It's not in a class so does it need that?
- nit: Can we change the param name to be plural `input_headers` just so its more obvious it contains all the headers.
- Can we add type hints for this? I know it's just a helper method but I'm trying to get into the habit of adding Python 3 style type hints for all new functions.

To sum it all up, something like this (assuming these are the right types)
```py
def _storage_header_sort(input_headers: List[Tuple(str, str)]) -> List[Tuple(str, str)]):
```

Reply:
1. Sounds good to me!
2. Nice catch! I originally had this as a class method then copied and pasted it out, removed 😄 
3. 👍 
4. 👍  will have to import `from Typing import List, Tuple` but I also think this is very helpful for future maintainers so worth it!